bot.js

```diff
@@ -2,12 +2,44 @@ require('dotenv').config();
 const { Telegraf, Scenes, session } = require('telegraf');
 const axios = require('axios');
 
+// Initialize the bot and use session middleware
 const bot = new Telegraf(process.env.TELEGRAM_BOT_TOKEN);
 bot.use(session());
 
 const HYGRAPH_API = process.env.HYGRAPH_API_URL;
 const HYGRAPH_API_TOKEN = process.env.HYGRAPH_API_TOKEN;
 
+// Rate-limiting configuration
+const rateLimit = {};
+const LIMIT = 5; // Maximum requests per user
```

Comment:
Maybe increase the limit a bit cus 5 seems a bit low , cus sometimes callbacks might take sometime